xclim/indices/_threshold.py

```diff
@@ -1280,143 +1213,164 @@ def frost_free_season_start(
     thresh : Quantified
         Threshold temperature on which to base evaluation.
     window : int
-        Minimum number of days with temperature above threshold needed for evaluation.
+        Minimum number of days with temperature above/under threshold to start/end the season.
+    mid_date : DayOfYearStr, optional
+        A date what must be included in the season.
+    op : {'>', '>=', 'ge', 'gt'}
+        How to compare tasmin and the threshold.
     freq : str
         Resampling frequency.
 
     Returns
     -------
     xarray.DataArray, [dimensionless]
-        Day of the year when minimum temperature is superior to a threshold
-        over a given number of days for the first time.
-        If there is no such day or if a frost free season is not detected, returns np.nan.
+        Day of the year when the frost free season starts.
 
     Notes
     -----
     Let :math:`x_i` be the daily mean temperature at day of the year :math:`i` for values of :math:`i` going from 1
-    to 365 or 366. The start date of the start of growing season is given by the smallest index :math:`i`:
+    to 365 or 366. The start date of the season is given by the smallest index :math:`i`:
 
     .. math::
 
        \prod_{j=i}^{i+w} [x_j >= thresh]
 
     where :math:`w` is the number of days the temperature threshold should be met or exceeded,
-    and :math:`[P]` is 1 if :math:`P` is true, and 0 if false.
+    and `i` must be earlier than `mid_date`.
     """
-    thresh = convert_units_to(thresh, tasmin)
-    over = tasmin >= thresh
-    out = over.resample(time=freq).map(rl.first_run, window=window, coord="dayofyear")
-    out.attrs.update(units="", is_dayofyear=np.int32(1), calendar=get_calendar(tasmin))
-    return out
+    return season(
+        tasmin,
+        thresh=thresh,
+        window=window,
+        op=op,
+        stat="start",
+        freq=freq,
+        mid_date=mid_date,
+        constrain=(">", ">="),
+    )
 
 
 @declare_units(tasmin="[temperature]", thresh="[temperature]")
 def frost_free_season_end(
     tasmin: xarray.DataArray,
     thresh: Quantified = "0.0 degC",
-    mid_date: DayOfYearStr = "07-01",
     window: int = 5,
+    mid_date: DayOfYearStr = "07-01",
+    op: str = ">=",
     freq: str = "YS",
 ) -> xarray.DataArray:
     r"""End of the frost free season.
```

Comment:
I don't want to stir the pot more... but why not optional `mid_date` here too? is it different from `growing_degree...` ?

Reply:
Indeed, I saw that too! It has been modified in the last commit.